signer/gpg2/gpg2_test.go

```diff
@@ -14,7 +15,14 @@ import (
 )
 
 func assertNewSignerWithConfOK(t *testing.T, conf signer.Configuration) *GPG2Signer {
-	s, err := New(conf)
+	// These test names end up quite long, and some versions of GPG don't work well
+	// with very long paths. See https://bugzilla.redhat.com/show_bug.cgi?id=1813705
+	// To work around this, we use a hashed version of the test name. This keeps
+	// the prefix consistent across runs while still being short enough.
+	hashedName := md5.Sum([]byte(t.Name()))
+	tmpDirPrefix := fmt.Sprintf("%x", hashedName)
+
+	s, err := New(conf, tmpDirPrefix)
```

Comment:
What if we passed in the full temp dir into New, instead? That would me we wouldn't have to glob for anything, we'd handle the edge cases where the base temp dir isn't `/tmp`, we'd know exactly what directory we can delete after the test is over and there wouldn't ever be a collision with a different test. (This is what I thought you were going to do, and may have misspoke and made you think I didn't want that)

Reply:
That it was I was originally thinking of doing, yeah. I backed off of it after I realized we had access to `tmpDir` after the fact, so it didn't seem necessary to do.

Reply:
Oh, oh, so we actually can use `s.tmpDir` in the `os.RemoveAll`?

Reply:
Which one are you talking about?

Reply:
The one at line 371?

Reply:
Nevermind, I clearly misread!